discord/poll.py

```diff
@@ -388,9 +388,6 @@ def _from_data(cls, *, data: PollPayload, message: Message, state: ConnectionSta
         # self.created_at = message.created_at
         # duration = self.created_at - expiry
 
-        if (duration.total_seconds() / 3600) > 168:  # As the duration may exceed little milliseconds then we fix it
-            duration = datetime.timedelta(days=7)
-
```

Comment:
Why are we removing this clamp instead of altering it to meet the new criteria?

Reply:
...and upon such time we would remove said clamp, we shouldn't be doing it for features we hope discord may release.

Reply:
The check is removed so polls aren't capped at 7 days when obtained via API or gateway, in my opinion updating this every time Discord updates the polls maximum duration would not be right.

Reply:
If that is the argument now then I wonder why it was implemented and accepted in the first place.

Reply:
It's unlikely discord ever adds an uncapped poll. The feature doesn't make sense. The point of a poll is getting an answer from a group of people, if it never ends, there's no answer ever.

Reply:
It's fine to remove the check and let the server handle it, that's what the library usually does anyway except within `discord.ui`